src/common/input/InputManager.cpp

```diff
@@ -354,8 +355,9 @@ bool InputDeviceManager::UpdateXboxPortInput(int usb_port, void* Buffer, int Dir
 		xid_type < to_underlying(XBOX_INPUT_DEVICE::DEVICE_MAX));
 	bool has_changed = false;
 
+	// First check if ImGui is focus, then ignore any input update occur.
 	// If somebody else is currently holding the lock, we won't wait and instead report no input changes
-	if (m_Mtx.try_lock()) {
+	if (!g_renderbase->IsImGuiFocus() && m_Mtx.try_lock()) {
```

Comment:
```suggestion
	if (static_cast<uint8_t>(!g_renderbase->IsImGuiFocus()) & static_cast<uint8_t>(m_Mtx.try_lock())) {
```

It's the same optimization I used in https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/pull/2152 to reduce the impact of branch misprediction.